isobus/src/nmea2000_message_definitions.cpp

```diff
@@ -0,0 +1,1134 @@
+		{
+			bool retVal = (messageTimestamp_ms != timestamp);
+			messageTimestamp_ms = timestamp;
+			return retVal;
+		}
+
+		std::uint16_t CourseOverGroundSpeedOverGroundRapidUpdate::get_raw_course_over_ground() const
+		{
+			return courseOverGround;
+		}
+
+		float CourseOverGroundSpeedOverGroundRapidUpdate::get_course_over_ground() const
+		{
+			return (10E-4f * courseOverGround);
+		}
+
+		bool CourseOverGroundSpeedOverGroundRapidUpdate::set_course_over_ground(std::uint16_t course)
+		{
+			bool retVal = (courseOverGround != course);
+			courseOverGround = course;
+			return retVal;
+		}
+
+		std::uint16_t CourseOverGroundSpeedOverGroundRapidUpdate::get_raw_speed_over_ground() const
+		{
+			return speedOverGround;
+		}
+
+		float CourseOverGroundSpeedOverGroundRapidUpdate::get_speed_over_ground() const
+		{
+			return (10E-2f * speedOverGround);
+		}
+
+		bool CourseOverGroundSpeedOverGroundRapidUpdate::set_speed_over_ground(std::uint16_t speed)
+		{
+			bool retVal = (speedOverGround != speed);
+			speedOverGround = speed;
+			return retVal;
+		}
+
+		std::uint8_t CourseOverGroundSpeedOverGroundRapidUpdate::get_sequence_id() const
+		{
+			return sequenceID;
+		}
+
+		bool CourseOverGroundSpeedOverGroundRapidUpdate::set_sequence_id(std::uint8_t sequenceNumber)
+		{
+			bool retVal = (sequenceID != sequenceNumber);
+			sequenceID = sequenceNumber;
+			return retVal;
+		}
+
+		CourseOverGroundSpeedOverGroundRapidUpdate::CourseOverGroudReference CourseOverGroundSpeedOverGroundRapidUpdate::get_course_over_ground_reference() const
+		{
+			return cogReference;
+		}
+
+		bool CourseOverGroundSpeedOverGroundRapidUpdate::set_course_over_ground_reference(CourseOverGroudReference reference)
+		{
+			bool retVal = (cogReference != reference);
+			cogReference = reference;
+			return retVal;
+		}
+
+		void CourseOverGroundSpeedOverGroundRapidUpdate::serialize(std::vector<std::uint8_t> &buffer) const
+		{
+			buffer.resize(CAN_DATA_LENGTH);
+
+			buffer.at(0) = sequenceID;
+			buffer.at(1) = (0xFC | static_cast<std::uint8_t>(cogReference));
+			buffer.at(2) = static_cast<std::uint8_t>(courseOverGround & 0xFF);
+			buffer.at(3) = static_cast<std::uint8_t>((courseOverGround >> 8) & 0xFF);
+			buffer.at(4) = static_cast<std::uint8_t>(speedOverGround & 0xFF);
+			buffer.at(5) = static_cast<std::uint8_t>((speedOverGround >> 8) & 0xFF);
+			buffer.at(6) = 0xFF; // Reserved
+			buffer.at(7) = 0xFF; // Reserved
+		}
+
+		bool CourseOverGroundSpeedOverGroundRapidUpdate::deserialize(const CANMessage &receivedMessage)
+		{
+			bool retVal = false;
+
+			if (CAN_DATA_LENGTH == receivedMessage.get_data_length())
+			{
+				retVal |= set_sequence_id(receivedMessage.get_uint8_at(0));
+				retVal |= set_course_over_ground_reference(static_cast<CourseOverGroudReference>(receivedMessage.get_uint8_at(1) & 0x03));
+				retVal |= set_course_over_ground(receivedMessage.get_uint16_at(2));
+				retVal |= set_speed_over_ground(receivedMessage.get_uint16_at(4));
```

Comment:
There seem to be a lot more of these floating-point comparisons going on, I don't mind leaving it as an 'open' issue for now as it seems it'll take a bit of time to fix, up to you hahah

Reply:
I'm not sure this is what's happening here, since we're storing the raw values in the class, and comparing the raw value - see here: 
![image](https://github.com/Open-Agriculture/AgIsoStack-plus-plus/assets/10929341/9e24414c-8f73-4b35-8622-9cab90462581)

In this case we're comparing u16s

Reply:
Ahh my bad, I made my assumption to quickly. The problem lies one line further down, the ```retVal |= set_timestamp(SystemTiming::get_timestamp_ms());``` always sets the `retVal` to true as it always changes the value to the current timestamp. Maybe we want it to be just ```set_timestamp(SystemTiming::get_timestamp_ms());``` and not let it touch the `retVal` variable

Reply:
Ah, yep, I see. I think that makes sense.